src/keywords/ref_.rs

```diff
@@ -1,60 +1,78 @@
 use super::{CompilationResult, Validate};
 use crate::{
     compilation::{compile_validators, CompilationContext, JSONSchema},
-    error::{error, ErrorIterator},
+    error::{error, ErrorIterator, ValidationError},
+    keywords::Validators,
 };
+use parking_lot::RwLock;
 use serde_json::Value;
 use url::Url;
 
 pub struct RefValidator {
     reference: Url,
+    /// Precopmuted validators.
+    /// They are behind a RwLock as is not possible to compute them
+    /// at compile time without risking infinite loops of references
```

Comment:
Do you mean the rust compilation or validator compilation? Could you please elaborate a bit here?

Reply:
Ah, I didn't completely read your comment in the PR